ci: use images from quay.io

nebril opened the pull request:
This change causes jenkins CI jobs to not build images and instead use
images built by PR github action

related: https://github.com/cilium/cilium/issues/14928

nebril committed:
ci: use quay.io images in ginkgo-kernel.Jenkinsfile

This change causes CI jobs to not build images and instead use
images built by PR github action

Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

aanm commented on jenkinsfiles/ginkgo-kernel.Jenkinsfile:
```diff
@@ -193,6 +174,19 @@ pipeline {
+                retry(15) {
+                    sleep(time: 60)
+                    sh 'curl --silent -f -lSL "https://quay.io/api/v1/repository/cilium/cilium-ci/tag/${ghprbActualCommit}/images"'
+                    sh 'curl --silent -f -lSL "https://quay.io/api/v1/repository/cilium/cilium-operator-ci/tag/${ghprbActualCommit}/images"'
```
s/`cilium-operator-ci`/`operator-ci`

aanm commented on jenkinsfiles/ginkgo-kernel.Jenkinsfile:
```diff
@@ -225,21 +219,12 @@ pipeline {
-                HUBBLE_RELAY_TAG = "latest"
+                CILIUM_IMAGE = "quay.io/cilium/cilium-ci"
+                CILIUM_TAG = "${ghprbActualCommit}"
+                CILIUM_OPERATOR_IMAGE= "quay.io/cilium/cilium-operator-ci"
```
same here

nebril committed:
ci: use quay.io images in gke builds

This change cause quay.io images built in github actions to be used
instead of the locally built images.

gke-specific registry scripts are deleted

Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

tklauser commented on jenkinsfiles/ginkgo-gke.Jenkinsfile:
```diff
@@ -63,11 +63,6 @@ pipeline {
         }
-        stage('Start docker registry for build') {
-            steps{
-                sh 'cd ${TESTDIR}/gke; ./start-registry.sh'
```
It looks like the `start-registry.sh` script is no longer used after this PR. Should it be removed as well?

tklauser commented on jenkinsfiles/ginkgo-gke.Jenkinsfile:
```diff
@@ -199,7 +182,6 @@ pipeline {
   }
     post {
         always {
-            sh 'cd ${TESTDIR}/gke; ./stop-registry.sh || true'
```
Same here, it looks like `stop-registry.sh` is now no longer used. Remove it as well?

aanm commented on jenkinsfiles/ginkgo-gke.Jenkinsfile:
```diff
@@ -78,26 +73,32 @@ pipeline {
-                        sh 'cd ${TESTDIR}; ./make-images-push-to-local-registry.sh $(gke/registry-ip.sh) ${TAG} "--no-cache"'
+                stage('Wait for images') {
+                    options {
+                        timeout(time: 10, unit: 'MINUTES')
```
The images with `-race` might take longer time before they are available so this timeout needs to be bumped to at least 20 minutes.

nebril replied:
Good catch.

aanm commented on jenkinsfiles/ginkgo-kernel.Jenkinsfile:
```diff
@@ -193,6 +175,19 @@ pipeline {
         }
+        stage ("Wait for images") {
+            options {
+                timeout(time: 10, unit: 'MINUTES')
```
same here

aanm commented on jenkinsfiles/ginkgo-kernel.Jenkinsfile:
```diff
@@ -193,6 +175,19 @@ pipeline {
+                timeout(time: 10, unit: 'MINUTES')
+            }
+            steps {
+                retry(15) {
```
retry 20

aanm commented on jenkinsfiles/ginkgo-gke.Jenkinsfile:
```diff
@@ -78,26 +73,32 @@ pipeline {
-                                }
-                            }
+                    steps {
+                        retry(15) {
```
retry 20
